db/migrate/20170425154145_add_sti_to_middlewares.rb

```diff
@@ -0,0 +1,67 @@
+class AddStiToMiddlewares < ActiveRecord::Migration[5.0]
+  class MiddlewareDatasource < ApplicationRecord
+    self.table_name = 'middleware_datasources'
+  end
+
+  class MiddlewareDeployment < ApplicationRecord
+    self.table_name = 'middleware_deployments'
+  end
+
+  class MiddlewareDomain < ApplicationRecord
+    self.table_name = 'middleware_domains'
+  end
+
+  class MiddlewareMessaging < ApplicationRecord
+    self.table_name = 'middleware_messagings'
+  end
+
+  class MiddlewareServer < ApplicationRecord
+    self.table_name = 'middleware_servers'
+  end
+
+  class MiddlewareServerGroup < ApplicationRecord
+    self.table_name = 'middleware_server_groups'
+  end
+
+  def up
+    add_column :middleware_datasources, :type, :string, :null => false, :default => ''
+    add_column :middleware_deployments, :type, :string, :null => false, :default => ''
+    add_column :middleware_domains, :type, :string, :null => false, :default => ''
+    add_column :middleware_messagings, :type, :string, :null => false, :default => ''
+    add_column :middleware_server_groups, :type, :string, :null => false, :default => ''
+    add_column :middleware_servers, :type, :string, :null => false, :default => ''
+
+    MiddlewareDatasource
+      .where(:type => '')
```

Comment:
No, but I'm preparing for the case where there are some entities in the future that are not from Hawkular and the migration is run again. Probably useless future-proofing, but it's just one line and might save us in the future.